src/vcpkg/base/system.cpp

```diff
@@ -221,6 +227,32 @@ namespace vcpkg
 #endif
     }
 
+    void print_environment_variables()
```

Comment:
```suggestion
    std::string get_environment_block()
```

and let the calling debug place do the write_unlocalized_Xxx?

That way we make it clear that the reason this is unlocalized is because it's behind the debugging flag.